pkg/url/url.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package url
+
+import (
+	"fmt"
+
+	"istio.io/istio/operator/version"
+)
+
+var (
+	baseVersion = version.OperatorVersionString
+
+	// ReleaseTar is a URL to download latest istio version from Github release
+	ReleaseTar = `https://github.com/istio/istio/releases/download/` + baseVersion + `/istio-` + baseVersion + `-linux-amd64.tar.gz`
+)
+
+// istio.io related URLs
+var (
+	// BaseURL for istio.io
+	BaseURL = "https://istio.io/"
+
+	// DocsURL is a base docs URL for istio.io
+	DocsURL = fmt.Sprintf("%s%s", BaseURL, "latest/docs/")
```

Comment:
Do we want this to work properly for release branches? If so, then `latest/docs/` would be `release/docs` (e.g., `v1.6/docs`) instead. For `master` it would also need to change BaseURL to `preliminary.istio.io`.

Reply:
I think it will be good to make it work for different releases. Not required for `master` because these URL's are mostly exposed to end-user through `istioctl` commands.

I have updated code to consider the tag version.

https://istio.io/v1.4/docs/ops/deployment/requirements/
https://istio.io/v1.5/docs/ops/deployment/requirements/
https://istio.io/v1.6/docs/ops/deployment/requirements/

`v1.6` will be redirected to `latest`. 

Not sure how to redirect an un-released version to `preliminary.istio.io`.

Reply:
@davidhauck mentioned that he could setup `istio.io/v1.7` (or maybe something like `istio.io/next`) to redirect to `preliminary.istio.io/latest`. Then you could just use that for the un-released version. David, do you think you can set that up soon, so @shamsher31 can use it?

Reply:
@frankbu sure, I'll look into doing that soon

Reply:
Any updates @davidhauck ?

Reply:
Here it is: https://github.com/istio/istio.io/pull/7935.

Reply:
Thanks @davidhauck .

un-released version eg: `v1.7` now redirects to `preliminary.istio.io`

https://istio.io/v1.7/docs/ops/deployment/requirements/